test/metabase/upload_test.clj

```diff
@@ -2059,3 +2065,11 @@
                (is (= (map second (rows-with-auto-pk [(csv/read-csv original-row)]))
                       (map rest (rows-for-table table)))))
              (io/delete-file file))))))))
+
+(driver/register! ::short-column-test-driver)
+(defmethod driver/column-name-length-limit ::short-column-test-driver [_] 10)
+
+(deftest unique-long-column-names-test
+  (let [original ["abcdefghijk" "abcdefghijklm" "abcdefgh_2_etc" "abcdefgh_3_xyz"]
+        expected [:abcdefghij   :a_c1819420     :abcdefgh_2      :abcdefgh_3]]
```

Comment:
Some test coverage with unicode characters wouldn't be so hard to add, and seems the most likely thing to forget about and break in the future

Reply:
I've made a note to update these tests, and the pre-existing ones, in a later PR on Monday. Will go nicely on top of the display stuff you worked on.

Reply:
Oh well, there are merge conflicts now anyway, if CI needs to run again I'll just update this before merging.